### ISSUE TITLE
React 15.x compatibility

### ISSUE THREAD

**Issue by ingro** [buildo/rc-datepicker#78]
Hello!

Rc-datepicker works fine with React 0.14.x even if it throws deprecation warnings here and there.
Anyway those warnings became errors on React 15.x.

Those warnings regards the use of `require('react/addons')` in `src/DatePicker.js` (which is a typo I guess) and the use of the function `React.findDOMNode` in `src/DatePickerInput.js` which has been moved to the dedicated `react-dom` package.

Moreover this is not needed anymore in React 0.14.x because `this.refs.datePickerInput` [returns the DOM node by default](https://facebook.github.io/react/blog/2015/10/07/react-v0.14.html#dom-node-refs).

So I guess it's needed a release that enforce React 0.14+, to gain also the compatibility with React 15.

I could create a pull request but those are breaking changes so I wanted to discuss here first.

Let me know if I can proceed or if you prefeer to follow another path!

Thanks.

**Comment by FrancescoCioria** [comment]
Hi @ingro, sorry for the delay in my answer but I was waiting for our internal meeting to discuss about the migration to React15.

As we all want to (slowly) migrate to the final version of React I was thinking to finally update our libraries (`rc-datepicker` too) to React15 and simply keep a different branch `react-0.13.3` to maintina support for our older projects.

short answer: I'll create a PR asap to update `rc-datepicker`

**Comment by ingro** [comment]
@FrancescoCioria no worries, I'm happy to hear that!

**Comment by FrancescoCioria** [comment]
@ingro, @giogonzo has made a PR for this: https://github.com/buildo/rc-datepicker/pull/79
He lint-up and cleaned-up a lot of code, in case you want to have a look the part interesting for you should all be in the first commit: https://github.com/buildo/rc-datepicker/pull/79/commits/4072a05c1bf02ad89b0cd7a7e7f7df3b1e4bdeb3

**Comment by ingro** [comment]
Seems good to me! :)

**Comment by ingro** [comment]
Sorry to comment again on this but the current version `4.0.0` has the wrong peer-depencies set in package.json: 
```
 "peerDependencies": {
    "react": "^0.14 || ^0.15",
    "react-dom": "^0.14 || ^0.15"
  }
```

but should be:

```
 "peerDependencies": {
    "react": "^0.14 || ^15",
    "react-dom": "^0.14 || ^15"
  },
```

to be compatible with React 15.x.

Since version 5 it's still in beta do you think it could be possible to release an update to the 4.0.0 to fix this? Thanks.

**Comment by FrancescoCioria** [comment]
>Since version 5 it's still in beta do you think it could be possible to release an update to the 4.0.0 to fix this? Thanks.

sure!

**Comment by FrancescoCioria** [comment]
fixed with https://github.com/buildo/rc-datepicker/commit/849aa00d0832becf7cefc11951e7c1d4b2acf13e and published as `v4.0.1`

**Comment by ingro** [comment]
Thanks again @FrancescoCioria !
